app/models/link.rb

```diff
@@ -48,7 +49,34 @@ def self.build(params)
 
 private
 
-  def reset_time_and_status
-    self.update_columns(status: nil, link_last_checked: nil)
+  def find_link_url
+    find_existing_link_url || find_existing_homepage_url
+  end
+
+  def find_existing_link_url
+    Link.where(url: self.url).distinct.first
```

Comment:
You could memoize this method and `find_existing_homepage_url`. It might save you some extra assignments later on in the code.